src/components/ProfileResetPassword/ProfileResetPassword.tsx

```diff
@@ -0,0 +1,216 @@
+            }}
+          >
+            <Text style={globalStyles.overline2}>CONTINUE</Text>
+          </Pressable>
+        </View>
+      </Modal>
+    );
+  };
+
+  const setNewPassword = () => {
+    return (
+      <Modal visible={visible} transparent animationType="slide">
+        <View style={styles.modalView}>
+          <View style={styles.topGreyLine} />
+          <Pressable style={styles.exitPressable} onPress={() => reset()}>
+            <SvgIcon type="greyX" />
+          </Pressable>
+          <Text style={[globalStyles.h2, styles.resetPasswordTitle]}>
+            Set New Password
+          </Text>
+          <TextInput
+            style={[globalStyles.body1, styles.setNewInput]}
+            placeholder="New Password"
+            placeholderTextColor="#A9A9A9"
+            onChangeText={(e) => [onChangeNewPW(e), setEdited(true)]}
+            value={newPW}
+          />
+          <Pressable style={styles.passwordHiddenIconSet}>
+            <SvgIcon type="passwordHidden" />
+          </Pressable>
+          <Text style={styles.mustBeCharacter}>
+            Must be 8-20 characters long
+          </Text>
+          <TextInput
+            style={[globalStyles.body1, styles.confirmNewInput]}
+            placeholder="Confirm New Password"
+            placeholderTextColor="#A9A9A9"
+            onChangeText={(e) => [onChangeConfirmPW(e), setEdited(true)]}
+            value={confirmPW}
+          />
+          <Pressable style={styles.passwordHiddenIconConfirm}>
+            <SvgIcon type="passwordHidden" />
+          </Pressable>
+          <Text style={styles.pwMustMatch}>Passwords must match</Text>
+          <Pressable
+            style={
+              edited && valid()
+                ? styles.verifyContinuePressableAllowed
+                : styles.verifyContinuePressable
+            }
+          >
+            <Text
+              style={globalStyles.overline2}
+              onPress={() => (valid() ? setCurrScreen("success") : null)}
+            >
+              RESET PASSWORD
+            </Text>
+          </Pressable>
+        </View>
+      </Modal>
+    );
+  };
+
+  const success = () => {
+    return (
+      <Modal visible={visible} transparent animationType="slide">
+        <View style={styles.modalView}>
+          <View style={styles.topGreyLine} />
+          <Pressable style={styles.exitPressable} onPress={() => reset()}>
+            <SvgIcon type="greyX" />
+          </Pressable>
+          <Text style={[globalStyles.h2, styles.resetPasswordTitle]}>
+            Success!
+          </Text>
+          <Text style={[globalStyles.h4, styles.subText1]}>
+            Your new password was successfully set.
+          </Text>
+          <Pressable style={styles.closePressable} onPress={() => reset()}>
+            <Text style={globalStyles.overline2}>CLOSE</Text>
+          </Pressable>
+        </View>
+      </Modal>
+    );
+  };
+
+  if (currScreen == "reset") {
```

Comment:
I personally like switch statements when there's a lot of cases like these. This is fine too, but I think switch statements are a lil cleaner? switch it only if u want!